flask_resty/authorization.py

```diff
@@ -5,29 +5,80 @@
 
 
 class AuthorizationBase(object):
+    """Base class for the API authorization scheme. Typically you will want
+    to use the :py:class:`AuthorizeModifyMixin` to implement uniform
+    authorization logic for each CRUD operation.
+
+    Implementing classes can customize the authorization scheme at multiple
+    points in an :py:class:`ApiView`:
+
+    * When requests are processed
+    * When ORM operations are called
+    * When queries are executed
+    """
     def get_request_credentials(self):
+        """Retrieves the credentials stored in the
+        :py:class:`flask.ctx.AppContext`.
+        """
         return authentication.get_request_credentials()
 
     def authorize_request(self):
+        """Authorize the current request. Typically this is done by inspecting
+        :py:data:`flask.request`.
+        """
         raise NotImplementedError()
 
     def filter_query(self, query, view):
+        """Apply a filter to the provided `query` using the model, schema or
+        other data from the provided `view`.
+
+        :param query: The SQL construction object.
+        :type query: :py:class:`sqlalchemy.orm.query.Query`
+        :param view: The View instance
+        :type view: :py:class:`ApiView`
+        :return: The filtered SQL construction object.
+        :rtype: :py:class:`sqlalchemy.orm.query.Query`
+        """
         raise NotImplementedError()
 
     def authorize_save_item(self, item):
+        """Authorize attempts to add a model instance to the current
+        :py:class:`sqlalchemy.orm.session.Session` or update a model instance
+        from the current :py:class:`sqlalchemy.orm.session.Session`.
+
+        :param obj item: The model instance
+        """
         raise NotImplementedError()
 
     def authorize_create_item(self, item):
+        """Authorize attempts to create a new instance of a model that
+        inherits from the declarative base class.
+
+        :param obj item: The model instance
+        """
         raise NotImplementedError()
 
     def authorize_update_item(self, item, data):
+        """Authorize attempts to update an existing instance of a model that
+        inherits from the declarative base class.
+
+        :param obj item: The model instance
+        :param dict data: A mapping from field names to updated values
+        """
         raise NotImplementedError()
 
     def authorize_delete_item(self, item):
+        """Authorize attempts to delete an existing instance of a model that
+        inherits from the declarative base class.
+
+        :param obj item: The model instance
+        """
         raise NotImplementedError()
 
 
 class HasCredentialsAuthorizationBase(AuthorizationBase):
+    """An authorization scheme where only authenticated requests are allowed.
```

Comment:
what is an "authenticated request"? I don't think that concept is defined here.

Reply:
I should qualify that this means "any value is set on the key "request_credentials" in the flask-resty context".